tests/wpunit/Tribe/Tickets/OrphanedPostsTest.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+namespace Tribe\Tickets;
+
+use TEC\Tickets\Commerce\Module;
+use Tribe\Tickets\Test\Commerce\Attendee_Maker;
+use Tribe\Tickets\Test\Commerce\TicketsCommerce\Ticket_Maker as TC_Ticket_Maker;
+use Tribe\Tickets\Test\Commerce\RSVP\Ticket_Maker as RSVP_Ticket_Maker;
+use Tribe__Tickets__Tickets as Tickets;
+
+/**
+ * Test class for orphaned posts functionality.
+ *
+ * @since TBD
+ */
+class OrphanedPostsTest extends \Codeception\TestCase\WPTestCase {
+
+	use RSVP_Ticket_Maker;
+	use TC_Ticket_Maker;
+	use Attendee_Maker;
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		// Enable post as ticket type.
+		add_filter( 'tribe_tickets_post_types', function () {
+			return [ 'post' ];
+		} );
+
+		// Enable Tickets Commerce.
+		add_filter( 'tribe_tickets_commerce_tc_is_active', '__return_true' );
+	}
+
+	/**
+	 * Test get_orphaned_post_ids with RSVP provider.
+	 *
+	 * @test
+	 */
+	public function should_get_orphaned_post_ids_for_rsvp_provider() {
+		// Create an event that will be deleted to create orphaned tickets.
+		$event_id = tribe_events()->set_args( [
+			'title'      => 'Test Event for Orphaned RSVP',
+			'status'     => 'publish',
+			'start_date' => '2025-01-01 00:00:00',
+			'duration'   => 2 * HOUR_IN_SECONDS,
+		] )->create()->ID;
+
+		// Create RSVP tickets for the event.
+		$rsvp_ticket_id = $this->create_rsvp_ticket( $event_id );
+
+		// Create attendees for the RSVP ticket.
+		$attendee_ids = $this->create_many_attendees_for_ticket( 3, $rsvp_ticket_id, $event_id );
+
+		// Verify tickets and attendees exist before deletion.
+		$this->assertNotEmpty( $rsvp_ticket_id );
+		$this->assertNotEmpty( $attendee_ids );
+
+		// Delete the event to create orphaned tickets/attendees.
+		wp_delete_post( $event_id, true );
+
+		// Get orphaned post IDs for RSVP provider.
+		$orphaned_post_ids = tribe( 'Tribe__Tickets__RSVP' )->get_orphaned_post_ids( 'Tribe__Tickets__RSVP' );
+
+		// Should find the orphaned RSVP ticket and attendees.
+		$this->assertContains( $rsvp_ticket_id, $orphaned_post_ids );
+		foreach ( $attendee_ids as $attendee_id ) {
+			$this->assertContains( $attendee_id, $orphaned_post_ids );
```

Comment:
In fully controlled environment, assertEquals is preferred. We know exactly what the results should be, so lets test for exactly that.